### PR TITLE
FIX 6864 - improve the loading of babel config

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybooks/storybook/issues/6864

## What I did
CHANGE load-custom-babel-config file so it will try to load as JS & then as JSON, and use whichever works

### CODE REVIEW

**Review by shilman — APPROVED** [review]
LGTM

File: lib/core/src/server/utils/load-custom-babel-config.js [hunk 1]
```diff
@@ -16,19 +16,37 @@ function removeReactHmre(presets) {
...
+      error.json = e;
     }
+
+    if (!config) {
+      logger.error(`=> Error parsing babel config file: ${babelConfigPath}
+      
+      We tried both loading as JS & JSON, neither worked.
+      Maybe there's a syntax error in the file?`);
+      logger.error(`=> From JS loading we got: ${error.js.message}`);
+      logger.error(`=> From JSON loading we got: ${error.js.message}`);
```
Comment (lydell) [inline lib/core/src/server/utils/load-custom-babel-config.js]: This line should say `error.json.message`, shouldn’t it?

Comment (shilman) [reply lib/core/src/server/utils/load-custom-babel-config.js]: good catch!

Comment (lydell) [reply lib/core/src/server/utils/load-custom-babel-config.js]: Sorry, I’m afraid not :(

**Pull request merged**